pkg/parser/parser.go

```diff
@@ -300,19 +306,34 @@ func (p *Parser) Schema() *schema.Schema {
 	return schema
 }
 
-func resolveDependenciesShallow(tasks []task.ParseableTaskLike) {
+func (p *Parser) resolveDependenciesShallow(tasks []task.ParseableTaskLike) error {
 	for _, task := range tasks {
 		var dependsOnIDs []int64
 		for _, dependsOnName := range task.DependsOnNames() {
+			// Dependency may be missing due to only_if
+			if _, ok := p.disabledTaskNamesAndAliases[dependsOnName]; ok {
+				break
```

Comment:
I think it should be `continue`. Seems you don't want to completely break.

Reply:
I think an test case like this will be incorrect:

```yaml
container:
  image: latest

foo_task:
  depends_on: 
    - bar
    - baz
  script: echo foo

bar_task:
  only_if: false
  script: echo bar

baz_task:
  script: echo baz
```

Reply:
Good catch! Fixed in 60923ac.